pkg/runctl/run_manager_test.go

```diff
@@ -85,6 +87,87 @@ func Test_RunManager_Start_DoesNotSetPipelineRunStatus(t *testing.T) {
 	mockPipelineRun.EXPECT().UpdateState(gomock.Any()).Times(0)
 }
 
+func Test_RunManager_Start_DoesCopySecret(t *testing.T) {
+	t.Parallel()
+
+	// SETUP
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	secretName := "scm_secret1"
+	spec := &steward.PipelineSpec{
+		JenkinsFile: steward.JenkinsFile{
+			Secret: secretName}}
+	mockFactory, mockPipelineRun, mockSecretProvider, mockNamespaceManager := prepareMocksWithSpec(mockCtrl, spec)
+
+	preparePredefinedSecrets(mockSecretProvider, secretName)
+	preparePredefinedClusterRole(t, mockFactory, mockPipelineRun)
+
+	examinee := NewRunManager(mockFactory, mockSecretProvider, mockNamespaceManager)
+
+	// VERIFY
+	// UpdateState should never be called by BuildStarter
+	mockPipelineRun.EXPECT().UpdateState(gomock.Any()).Do(func(interface{}) { panic("unexpected call") }).AnyTimes()
```

Comment:
This is not a verification section but the definition of a test stub that belongs to the setup section.